### PR TITLE
CORE-5166 Improve logging in the avro serializer

### PR DESCRIPTION
**Pull request by simon-johnson-r3**

### CODE REVIEW

File: libs/messaging/kafka-message-bus-impl/src/test/kotlin/net/corda/messagebus/kafka/serialization/CordaAvroSerializerImplTest.kt [hunk 1]
```diff
@@ -39,4 +42,16 @@ class CordaAvroSerializerImplTest {
     fun testByteArrayValue() {
         assertThat(cordaAvroSerializer.serialize(topic, "bytearray".toByteArray()) != null)
     }
+
+    @Test
+    fun testCustomClassValue() {
```
Comment (simon-johnson-r3) [inline libs/messaging/kafka-message-bus-impl/src/test/kotlin/net/corda/messagebus/kafka/serialization/CordaAvroSerializerImplTest.kt:47]: This is not related to this change, except whilst making it I discovered without this test the setup at line 23 is redundant, which also means the ensuing code path was never tested. This is because the avro schema registry is never used to serialize strings and byte buffers which is the target of the other tests.

File: libs/messaging/kafka-message-bus-impl/build.gradle [hunk 2]
```diff
@@ -30,5 +30,6 @@ dependencies {
     testImplementation project(":testing:test-utilities")
 
     testRuntimeOnly 'org.osgi:osgi.core'
+    testRuntimeOnly "org.apache.logging.log4j:log4j-slf4j-impl:$log4jVersion"
```
Comment (simon-johnson-r3) [inline libs/messaging/kafka-message-bus-impl/build.gradle:33]: These are added only so I could test the output of the logging made sense and didn't itself result in any errors.

**Review by JamesHR3 — APPROVED** [review]
